### DISCUSSION TITLE
Authenticate with basic-auth without redirect

### DISCUSSION THREAD

**Discussion by rfx77** [OpenIDC/mod_auth_openidc#540] [General]
Hi!

I want to protect a service with keycloak. the current clients do a basic-authentication against apache. when i change to mod_auth_openidc the simple basic-auth header does not authenticate anymore. i get redirects to the sso server. even if i could get it to wirk with curl with follow-redirects and location-trust and cookies the current clients do not support redirects.

is it possible to implement the same behavior as before where i can use a curl with -u user:pass to access the service which is protected. Bearer or token is not an option and i know how this would be done if i could implement it myself but i cannot change the client.

Greetings,
Franz

**Comment by zandbelt** [comment]
I assume you want to support both behaviours at the same time? That is not possible unless you create different URLs/Locations for both clients.

**Reply by rfx77** [reply]
Supporting both is not needed. i only want basic-auth

**Reply by zandbelt** [reply]
well, then you should not use this module but stick with the old configuration, or am I missing something?

**Reply by rfx77** [reply]
th backend-service should get random passwords which are saved in a keycloak attribute and athe user does not know but thats only a detail of my implementation. what i simply want is basic auth against apache with mod_auth_openidc protected locations and how i can configure this.

**Comment by rfx77** [comment]
It should work the same way as "enable-basic-auth" in the KeyCloak Java Adapter
